### ISSUE TITLE
(Jupyter Extension) Customers failing on remote SSH connection due to missing file

### ISSUE THREAD

**Issue by IanMatthewHuff** [microsoft/vscode-remote-release#4479]
I'm a developer on the Jupyter Extension.

Just recently we've had a number of reports come in about customers having issues with using our extension over Remote - SSH.

https://github.com/microsoft/vscode-jupyter/issues/4698

I'm not familiar with exactly what the Remote-SSH extension does to move the files over to the remote machine, but at first glance it seems that one particular file is not getting moved over the remote machine. Is there more that we can do to diagnose this? I'm not sure if this is a red herring or not, but that one file does have a very long, involved file name.

**Comment by bamurtaugh** [comment]
Thanks for reporting @IanMatthewHuff. If the file/folder is large, I'd think that could play a role, i.e. https://github.com/microsoft/vscode-remote-release/issues/1257. 

Is there any output you/the customers could share from the Remote - SSH output logs when this happens?

cc @roblourens

**Comment by roblourens** [comment]
There isn't any moving files around here. Remote extensions are downloaded from the marketplace and unzipped in pretty much the same way as when they are installed locally.

@IanMatthewHuff I don't exactly understand what's happening in that issue, is there a file missing from the installed extension? Which file? Can you have them try reinstalling the extension?

**Comment by IanMatthewHuff** [comment]
@roblourens and @bamurtaugh My apologies for the slow reply. I was out of the office for a bit. Currently I'm still working at getting a lockdown on the issue. What I have currently is that a number of customers over the last threeish weeks (this was over a timeframe where we didn't update our extension at all) have started running into issues with using our extension with remote-ssh. What it looks like, is that on the remote machine a specific file is missing or unable to be found. 

Part of the reason for looping you in on this issue is that I'm a bit unsure of how to debug what seems to be a missing file from the remote extension folder. I don't use remote-ssh myself, so I'm going to take a stab at setting this up to investigate next and I've ask the customers in the thread in our repo to pull the Remote - SSH logs as next steps.

**Comment by roblourens** [comment]
No worries. We've had cases like this before where some antivirus quarantined a file. I see there are Linux users in that issue so maybe that's less likely. But I would look out for unusual external factors like that.

**Comment by IanMatthewHuff** [comment]
@roblourens Thanks for the info. No clue if this is a red herring or not, but the only thing that seems unusual or different about that file is that it has an extremely long file name (it's pulled in from an external dependency, so we don't control the name). Since you mentioned antivirus I'm not sure if something like that would make it more likely to be flagged.
`vendors~dataresource~geojson~modeldebug~nteract_transforms~nteract_transforms_vsdom.bundle.js`

**Comment by IanMatthewHuff** [comment]
@roblourens  and @bamurtaugh . Been working with some customer logs on this issue. First thing that I found was that in a failure log there was another extension that was throwing an exception. So I had the customer disable that extension and try again:
https://github.com/microsoft/vscode-jupyter/issues/4698#issuecomment-792983148

However the customer was still getting failures after that. I've attached a log from them to this comment.
[failed.txt](https://github.com/microsoft/vscode-remote-release/files/6125353/failed.txt)

There seem to be some errors in this log, but I'm not sure which ones are relevant. 

`[19:35:31.414] Resolver error: Error: Connecting with SSH timed out
	at Function.Timeout (c:\Users\Bowen\.vscode\extensions\ms-vscode-remote.remote-ssh-0.65.1\out\extension.js:1:64454)
	at Timeout._onTimeout (c:\Users\Bowen\.vscode\extensions\ms-vscode-remote.remote-ssh-0.65.1\out\extension.js:1:295174)
	at listOnTimeout (internal/timers.js:549:17)
	at processTimers (internal/timers.js:492:7)`

Looks like that might be an error that was passed on retries. At least from my reading of the log.

`[19:35:44.315] Failed to set up socket for dynamic port forward to remote port 43083: Proxy connection timed out. Is the remote port correct?`

Does this log look like everything was working correct from the ssh-remote perspective?
